src/pydvl/influence/array.py

```diff
@@ -0,0 +1,390 @@
+            return_stored: If True, returns the stored Zarr data object. Defaults to False.
+            overwrite: If True, overwrites any existing data at the specified location.
+
+
+        Returns:
+            Depending on the value of return_stored, this method may return a reference to the stored
+            Zarr data object or None.
+        """
+
+
+class OneDimChunkedLazyArrayProvider(LazyArrayProvider):
+    """
+    A class representing chunked, and lazily evaluated array, where the chunking is restricted to the first dimension
+
+    This class is designed to handle large arrays that are not feasible to load into memory all at once.
+    It works by generating chunks of the array on demand and can also convert these chunks to a Zarr array
+    for efficient storage and retrieval.
+
+    Attributes:
+        generator_factory: A factory function that returns
+            a generator. This generator yields chunks of the large array when called.
+    """
+
+    def __init__(
+        self, generator_factory: Callable[[], Generator[TensorType, None, None]]
+    ):
+        self.generator_factory = generator_factory
+
+    def compute(self, tensor_aggregator: Optional[TensorAggregator] = None):
+        """
+        Computes and optionally aggregates the chunks of the array using the provided tensor_aggregator.
+        This method initiates the generation of chunks and then combines them according to the aggregator's logic.
+
+        Args:
+            tensor_aggregator: An optional aggregator for combining the chunks of
+                the array. If None, a default ListAggregator is used to simply collect the chunks into a list.
+
+        Returns:
+            The aggregated result of all chunks of the array, the format of which depends on the tensor_aggregator used.
+
+        """
+        if tensor_aggregator is None:
+            tensor_aggregator = ListAggregator()
+        return tensor_aggregator.aggregate_from_generator(self.generator_factory())
+
+    def to_zarr(
+        self,
+        path_or_url: str,
+        numpy_converter: NumpyConverter,
+        return_stored: bool = False,
+        overwrite: bool = False,
+    ):
+        """
+        Converts the array into Zarr format, a storage format optimized for large arrays, and stores it at
+        the specified path or URL. This method is suitable for scenarios where the data needs to be saved
+        for later use or for large datasets requiring efficient storage.
+
+        Args:
+            path_or_url: The file path or URL where the Zarr array will be stored.
+            numpy_converter: A converter for transforming blocks into NumPy arrays compatible with Zarr.
+            return_stored: If True, the method returns the stored Zarr array; otherwise, it returns None.
+            overwrite: If True, overwrites existing data at the given path_or_url. If False, an error
```

Comment:
Formatting of variable names and values (here and elsewhere)
```suggestion
            overwrite: If `True`, overwrites existing data at the given `path_or_url`. If `False`, an error
```